main.py

```diff
@@ -148,8 +148,16 @@ def main(args: argparse.Namespace) -> None:
                     if sensor == 'LIDAR_TOP':
                         # nuscenes lidar data is stored as (x, y, z, intensity, ring index)
                         scan = np.fromfile(file_path, dtype=np.float32)
-                        points = scan.reshape((-1, 5))
-                        data = {"x": points[:, 0], "y": points[:, 1], "z": points[:, 2]}
+                        points = scan.reshape((-1, 5)).T
+
+                        # Transform lidar points from point sensor frame to egovehicle frame
+                        calibration = nusc.get('calibrated_sensor', sensor_data['calibrated_sensor_token'])
+                        rot_matrix = Quaternion(calibration['rotation']).rotation_matrix
+                        points[:3, :] = np.dot(rot_matrix, points[:3, :])
```

Comment:
looks good. To avoid all of this R,t boilerplate, the abstraction I like to use is `argoverse.se3.SE3`. I try to follow the convention [here](https://github.com/argoai/argoverse-api#a-note-regarding-coordinate-transforms) so that the coordinate frames are always obvious. I would prefer to write
```python
from argoverse.utils.se3 import SE3
from argoverse.utils.transform import quat2rotmat
points_lidar = np.fromfile(file_path, dtype=np.float32)
points_lidar = points_lidar.reshape((-1, 5)).T
egovehicle_R_lidar = quat2rotmat(calibration['rotation'])
egovehicle_t_lidar = calibration['translation']
egovehicle_SE3_lidar = SE3(rotation=egovehicle_R_lidar, translation=egovehicle_t_lidar)
points_egovehicle = egovehicle_SE3_lidar.transform_point_cloud(points_lidar)
data = {"x": points_egovehicle[:,0], "y": points_egovehicle[:,1], "z": points_egovehicle[:,2]}
```

Reply:
[quat2rotmat](https://github.com/argoai/argoverse-api/blob/master/argoverse/utils/transform.py#L14) expects a specific quaternion order (w, x, y, z). Which order does nuScenes use?

Reply:
Looks like `qw, qx, qy, qz = box.orientation.elements` is the order below, so should work fine.

Following the SE(3) notation could make things easier for the cuboid label stuff as well. Instead of

```python
box = Box(annotation['translation'], annotation['size'], Quaternion(annotation['rotation']))
box.translate(-np.array(ego_pose['translation']))
box.rotate(Quaternion(ego_pose['rotation']).inverse)
```
you could do
```python
city_SE3_object = SE3(annotation['rotation'], annotation['translation'])
city_SE3_egovehicle = SE3(translation=ego_pose['translation'],rotation=ego_pose['rotation'])
egovehicle_SE3_city = city_SE3_egovehicle.inverse()
egovehicle_SE3_object = egovehicle_SE3_city.right_multiply(city_SE3_object) # compose together
```

Reply:
Thanks for reviewing. SmalIt uses (w, x, y, z). 
```
calibrated_sensor {
   "token":                   <str> -- Unique record identifier.
   "sensor_token":            <str> -- Foreign key pointing to the sensor type.
   "translation":             <float> [3] -- Coordinate system origin in meters: x, y, z.
   "rotation":                <float> [4] -- Coordinate system orientation as quaternion: w, x, y, z.
   "camera_intrinsic":        <float> [3, 3] -- Intrinsic camera calibration. Empty for sensors that are not cameras.
}
```
It looks like this:
![image](https://user-images.githubusercontent.com/15787503/97014665-a4032500-1518-11eb-8328-3d8b31c8a434.png)